core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -1064,6 +1021,24 @@ public void shutdown() {
         result.mergeWith(dataAccess.deleteMetricData(id).flatMap(r -> null));
 
         result.mergeWith(dataAccess.deleteMetricFromRetentionIndex(id).flatMap(r -> null));
+        result.mergeWith(dataAccess.deleteFromMetricExpirationIndex(id).flatMap(r -> null));
+
         return result;
     }
+
+    @Override
+    public <T> void updateMetricExpiration(Metric<T> metric) {
```

Comment:
Let's keep things reactive and functional and return either Observable<Void> or (preferably) Completable instead of void.